core/src/graph/graph.rs

```diff
@@ -620,33 +635,37 @@ impl Graph {
 		page: &mut GraphPage,
 		connection_id: &DsnpUserId,
 		aggressive: bool,
-		keys: &Option<ResolvedKeyPair>,
 		dsnp_version_config: Option<&DsnpVersionConfig>,
-	) -> Result<()> {
+	) -> Result<(), AddConnectionError> {
 		let connection_type = self.get_connection_type();
 		let max_connections_per_page =
 			*PAGE_CAPACITY_MAP.get(&connection_type).unwrap_or_else(|| {
 				let mut capacities: Vec<&usize> = PAGE_CAPACITY_MAP.values().collect();
 				capacities.sort();
 				capacities.first().unwrap() // default: return smallest capacity value
 			});
+		let encryption_key = match self.get_connection_type().privacy_type() {
```

Comment:
why duplicate it here if we already had access to it? It this for ease of testing?

Reply:
Yes; made it easier for testing. Now that I've got that all sorted, I could factor this out and pass in if we want.